braindecode/models/eegitnet.py

```diff
@@ -0,0 +1,234 @@
+# Authors: Ghaith Bouallegue <ghaithbouallegue@gmail.com>
+#
+# License: BSD-3
+import torch
+from torch import nn
+
+from .modules import Ensure4d, Expression
+
+
+class _DepthwiseConv2d(torch.nn.Conv2d):
+    def __init__(
+            self,
+            in_channels,
+            depth_multiplier=2,
+            kernel_size=3,
+            stride=1,
+            padding=0,
+            dilation=1,
+            bias=True,
+            padding_mode="zeros",
+    ):
+        out_channels = in_channels * depth_multiplier
+        super().__init__(
+            in_channels=in_channels,
+            out_channels=out_channels,
+            kernel_size=kernel_size,
+            stride=stride,
+            padding=padding,
+            dilation=dilation,
+            groups=in_channels,
+            bias=bias,
+            padding_mode=padding_mode,
+        )
+
+
+
+class _InceptionBlock(nn.Module):
+    def __init__(self, in_ch, out_ch, kernel_sz, in_channels):
+        super().__init__()
+        self.inception_block = nn.Sequential(
+            *[
+                nn.Conv2d(
+                    in_ch, out_ch, kernel_size=(1, kernel_sz), padding="same", bias=False
+                ),
+                nn.BatchNorm2d(out_ch),
+                _DepthwiseConv2d(
+                    out_ch,
+                    kernel_size=(in_channels, 1),
+                    depth_multiplier=1,
+                    bias=False,
+                    padding="valid",
+                ),
+                nn.BatchNorm2d(out_ch),
+                nn.ELU(),
+            ]
+        )
```

Comment:
This code can be moved to `EEGITNet` class:

```suggestion
    @staticmethod
    def _get_inception_branch(in_channels, out_channels, kernel_length, depth_multiplier=1):
        return nn.Sequential(
            nn.Conv2d(
                1, out_channels, kernel_size=(1, kernel_length), padding="same", bias=False
            ),
            nn.BatchNorm2d(out_channels),
            _DepthwiseConv2d(
                out_channels,
                kernel_size=(in_channels, 1),
                depth_multiplier=depth_multiplier,
                bias=False,
                padding="valid",
            ),
            nn.BatchNorm2d(out_channels),
            nn.ELU(),
        )
```

Reply:
done